### PR TITLE
INT B-20706 MilMove Administrator Updates (Super Admin)

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

Sup you cool cats & kittens - we are at the beginning of creating **SUPER ADMINS**. When working on this, my hatred toward the Admin app grew even more because for some unknown reason, Truss decided to not use Redux or any state management for the Admin app.

Once I got the go ahead from Dre - I was able to add a simplified state to the store in the Admin application, which will allow us to check if the user that's currently logged into the Admin app is a **SUPER ADMIN**.

But to add to this fun after adding in some state objects, I also found out that there's not an endpoint that exists that we return the data in the `admin_users` table for the currently logged in user (using session data) - similar to the `getLoggedInUser` endpoint in the internal API.

The AC is kinda fun for this and these are the requirements:
- **SUPER ADMINS** can assign Safety Moves privileges to office users - NON supers cannot
- **SUPER ADMINS** cannot unsuper themselves (this was confirmed in Slack)
- **SUPER ADMINS** can see **SUPER ADMIN** data for other admins - NON supers cannot
- **SUPER ADMINS** can create other **SUPER ADMINS** - NON supers cannot

This PR does the following:
- adds migration to add the `super` column to the `admin_users` table
- implements Redux for the admin app and stores the `adminUser` and `oktaUser` objects - added a selector to retrieve admin info in the UI
- adds in the `/user` endpoint to the `admin` API that can be called and returns the currently signed in admin user's data
- updates existing endpoints to return the `super` attribute
- connects all relevant UI pages to the Redux store to retrieve the `adminUser` data
- added tests and cleaned up some old code

### How to test

1. Access MM as an Admin
2. You'll need to create an admin user or update an existing admin user and manually update the `super` column in the `admin_users` table to `true`
3. Navigate the admin app and confirm you can see the `Super Admin` attribute when viewing the list of admin users, selecting the admin user, and can update that value for admin users
4. Find an office user and confirm you can see and update the `Safety Moves` checkbox for that user
5. Try a non-super admin user and confirm you can't do any of the things in steps 3 or 4

### Super Admins can see `Super Admin` status in the list, on the quick view, as well as when editing admin profiles
Screenshot 2024-08-19 at 11 27 26 AM

Screenshot 2024-08-19 at 11 27 40 AM

Screenshot 2024-08-19 at 11 27 50 AM

### Non-Super Admins do not see all that fluff
Screenshot 2024-08-19 at 11 28 36 AM

Screenshot 2024-08-19 at 11 28 46 AM

Screenshot 2024-08-19 at 11 28 54 AM

### Supers can also update **Safety Moves** privileges for office users
Screenshot 2024-08-19 at 11 29 23 AM

### Non-Supers cannot (hidin' it)
Screenshot 2024-08-19 at 11 29 07 AM

### CODE REVIEW

File: src/pages/Admin/AdminUsers/AdminUserEdit.jsx [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+const AdminUserSuperAttribute = ({ adminUser }) => {
+  const record = useRecordContext();
+  // Hide the input so the super admin can't un-super themselves
+  if (record.id === adminUser.id) {
+    return null;
+  }
+  return (
+    <SelectInput
+      source="super"
+      choices={[
```
Comment (TevinAdams) [inline src/pages/Admin/AdminUsers/AdminUserEdit.jsx:22]: Just a suggestion. Could extract this array to its own constant.

File: src/utils/api.js [hunk 2]
```diff
@@ -38,6 +38,13 @@ export async function GetIsLoggedIn() {
   return response.body;
 }
 
+export async function GetAdminUser() {
+  const client = await getAdminClient();
+  const response = await client.apis.User.getLoggedInAdminUser({});
+  checkResponse(response, 'failed to get user due to server error');
```
Comment (traskowskycaci) [inline src/utils/api.js:44]: Small request to make this "failed to get _admin_ user due to server error"

File: src/shared/Entities/reducer.js [hunk 3]
```diff
@@ -81,3 +86,19 @@ export function entitiesReducer(state = initialState, action) {
 
   return state;
 }
+
+export function adminEntitiesReducer(state = adminInitialState, action) {
```
Comment (TevinAdams) [inline src/shared/Entities/reducer.js:90]: Is there a reason adminEntitiesReducer is not in its own file?

Comment (danieljordan-caci) [reply src/shared/Entities/reducer.js:90]: It's in this file with the other reducers

**Review by traskowskycaci — COMMENTED** [review]
Looks good functionally! Had one very small request

File: src/utils/api.js (hunk 2 above)
Comment (danieljordan-caci) [reply src/utils/api.js:44]: You dirty dog

Comment (danieljordan-caci) [reply src/utils/api.js:44]: You got it

Comment (traskowskycaci) [reply src/utils/api.js]: Thank you sir!

Comment (danieljordan-caci) [reply src/utils/api.js]: Done!

**Review by traskowskycaci — APPROVED** [review]
Looks good!

**Review by TevinAdams — APPROVED** [review]

**Pull request merged**